### PR TITLE
add ectyper module

### PR DESCRIPTION
**Pull request by rpetit3**
## PR checklist

Closes https://github.com/nf-core/modules/issues/942 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/ectyper/main.nf [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    """
+    if [ "$is_compressed" == "true" ]; then
+        gzip -c -d $fasta > $fasta_name
+    fi
+
+    ectyper \\
+        $options.args \\
+        --cores $task.cpus \\
+        --output ./ \\
+        --input $fasta_name
```
[the remark this answers is not in the document]
Comment (rpetit3) [reply modules/ectyper/main.nf]: Not a fan of it 
```
Output_directory is /home/runner/pytest_workflow_7emi2in0/ectyper_test_ectyper/work/d6/23fbfda94d19dad7ff02268892e309/./
Command-line arguments Namespace(input='/dev/fd/63', cores=2, percentIdentityOtype=90, percentIdentityHtype=95, percentCoverageOtype=90, percentCoverageHtype=50, verify=False, output='./', refseq=None, sequence=False, debug=False, dbpath=None)
RefSeq sketch (refseq.genomes.k21s1000.msh) and assembly meta data (assembly_summary_refseq.txt) is in good health and does not need to be downloaded
Gathering genome files
Using genomes in file /dev/fd/63
Identifying genome file types
/dev/fd/63 is not a fasta/fastq file
Folowing files were not found in the input: 
Creating combined serotype and identification fasta file
Assembling final list of fasta files
Non fasta / fastq file.
Standardizing the E.coli genome headers based on file names
Reporting results:
Name	Species	O-type	H-type	Serotype	QC	Evidence	GeneScores	AlleleKeys	GeneIdentities(%)	GeneCoverages(%)	GeneContigNames	GeneRanges	GeneLengths	Database	Warnings
63	-	-	-	-:-	-	-	-							v1.0 (11-03-2020)	Non fasta / fastq file. 

ECTyper has finished successfully.
```

[the remark this answers is not in the document]
Comment (grst) [reply modules/ectyper/main.nf:43]: :shrug: 

I guess we leave it as is then

File: modules/ectyper/main.nf [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    input:
+    tuple val(meta), path(fasta)
+
+    output:
+    tuple val(meta), path("*.log"), emit: log
+    tuple val(meta), path("*.tsv"), emit: tsv
+    tuple val(meta), path("*.txt"), emit: txt
+    path "versions.yml"              , emit: versions
```
Comment (lescai) [inline modules/ectyper/main.nf:28]:
```suggestion
    path "versions.yml"           , emit: versions
```

Comment (lescai) [reply modules/ectyper/main.nf:28]: minor thing, fix indent

**Review by lescai — APPROVED** [review]
just a minor indent formatting. but looks good to me otherwise.

**Comment by drpatelh** [comment]
Something 🐟 going on with the tests here too...

**Comment by rpetit3** [comment]
Yep! Another instance of me playing around with the recommended methods to use with comprossed inputs for tools that only support uncompressed inputs

**Comment by rpetit3** [comment]
@drpatelh I think I've fixed this, but honestly no idea why the lint is failing

**Pull request merged**